common/src/models/error.rs

```diff
@@ -30,114 +31,107 @@ impl Display for ApiError {
-                }
-            },
-            ErrorKind::ProjectNotReady => (
-                StatusCode::SERVICE_UNAVAILABLE,
-                // "not ready" is matched against in cargo-shuttle for giving further instructions on project deletion
-                "Project not ready. Try running `cargo shuttle project restart`.",
-            ),
-            ErrorKind::ProjectUnavailable => {
-                (StatusCode::BAD_GATEWAY, "Project returned invalid response")
-            },
-            ErrorKind::TooManyProjects => {
-                (StatusCode::FORBIDDEN, "You cannot create more projects. Delete some projects first.")
-            },
-            ErrorKind::ProjectHasRunningDeployment => (
-                StatusCode::INTERNAL_SERVER_ERROR,
-                "Could not automatically stop the running deployment for the project. Please reach out to Shuttle support for help."
-            ),
-            ErrorKind::ProjectHasBuildingDeployment => (
-                StatusCode::BAD_REQUEST,
-                "Project currently has a deployment that is busy building. Use `cargo shuttle deployment list` to see it and wait for it to finish"
-            ),
-            ErrorKind::ProjectCorrupted => (
-                StatusCode::BAD_REQUEST,
-                "Tried to get project into a ready state for deletion but failed. Please reach out to Shuttle support for help."
-            ),
+        let status = match kind {
+            ErrorKind::Internal => StatusCode::INTERNAL_SERVER_ERROR,
+            ErrorKind::KeyMissing => StatusCode::UNAUTHORIZED,
+            ErrorKind::ServiceUnavailable => StatusCode::SERVICE_UNAVAILABLE,
+            ErrorKind::KeyMalformed => StatusCode::BAD_REQUEST,
+            ErrorKind::BadHost => StatusCode::BAD_REQUEST,
+            ErrorKind::UserNotFound => StatusCode::NOT_FOUND,
+            ErrorKind::UserAlreadyExists => StatusCode::BAD_REQUEST,
+            ErrorKind::ProjectNotFound(_) => StatusCode::NOT_FOUND,
+            ErrorKind::ProjectNotReady => StatusCode::SERVICE_UNAVAILABLE,
+            // "not ready" is matched against in cargo-shuttle for giving further instructions on project deletion
+            ErrorKind::ProjectUnavailable => StatusCode::BAD_GATEWAY,
+            ErrorKind::TooManyProjects => StatusCode::FORBIDDEN,
+            ErrorKind::ProjectHasRunningDeployment => StatusCode::INTERNAL_SERVER_ERROR,
+            ErrorKind::ProjectHasBuildingDeployment => StatusCode::BAD_REQUEST,
+            ErrorKind::ProjectCorrupted => StatusCode::BAD_REQUEST,
             ErrorKind::ProjectHasResources(resources) => {
                 let resources = resources.join(", ");
                 return Self {
                     message: format!("Could not automatically delete the following resources: {}. Please reach out to Shuttle support for help.", resources),
                     status_code: StatusCode::INTERNAL_SERVER_ERROR.as_u16(),
```

Comment:
I think the debug formatting in the error macro is good enough, so this shouldn't be needed.